pl_bolts/datamodules/binary_emnist_datamodule.py

```diff
@@ -0,0 +1,180 @@
+        **kwargs: Any,
+    ) -> None:
+        """
+        Args:
+            data_dir (string, optional): Where to save/load the data.
+            split (string): The dataset has 6 different splits: ``byclass``, ``bymerge``,
+                            ``balanced``, ``letters``, ``digits`` and ``mnist``.
+                            This argument specifies which one to use.
+            val_split (int, float, None): Percent (float) or number (int) of samples
+                                          to use for the validation split.
+                                          If ``val_split = None``, the default size
+                                          (as in the EMNIST documentation) of
+                                          validation dataset is disbursed based on the ``split``.
+            num_workers (int): How many workers to use for loading data
+            normalize (bool): If ``True``, applies image normalize.
+            batch_size (int): How many samples per batch to load.
+            seed (int): Random seed to be used for train/val/test splits.
+            shuffle (bool): If ``True``, shuffles the train data every epoch.
+            pin_memory (bool): If ``True``, the data loader will copy Tensors into
+                               CUDA pinned memory before returning them.
+            drop_last (bool): If ``True``, drops the last incomplete batch.
+        """
+        if not _TORCHVISION_AVAILABLE:  # pragma: no cover
+            raise ModuleNotFoundError(
+                'You want to use EMNIST dataset loaded from `torchvision` which is not installed yet.'
+            )
+
+        super(BinaryEMNISTDataModule, self).__init__(  # type: ignore[misc]
+            data_dir=data_dir,
+            val_split=val_split,
+            num_workers=num_workers,
+            normalize=normalize,
+            batch_size=batch_size,
+            seed=seed,
+            shuffle=shuffle,
+            pin_memory=pin_memory,
+            drop_last=drop_last,
+            *args,
+            **kwargs,
+        )
+        self.split = split
+        self._check_and_update()
+
+    def _check_and_update(self):
+        if self._emnist_split_exists:
+            self._update_val_split()
+
+    def _update_val_split(self):
+        if (self.val_split is None):
+            if self.split_metadata.get('validation'):
+                self.val_split = int(self.split_metadata.get('num_test'))
+            else:
+                self.val_split = self._DEFAULT_NO_VALIDATION_VAL_SPLIT
+
+    @property
+    def _emnist_split_exists(self) -> bool:
+        """Checks if the split exists in emnist or not."""
+        if not any(s == self.split for s in self.dataset_cls.splits):
+            allowed_values = ', '.join([f'"{v}"' for v in self.dataset_cls.splits])
+            raise ValueError(
+                f'Invalid value provided for split (="{self.split}"). ' + f'Allowed splits are: {allowed_values}'
+            )
+        return True
+
```

Comment:
I don't think we need to check if a given `split` is valid as torchvision does it anyway at https://github.com/pytorch/vision/blob/015378434c1e4be778b2d65eb86f227db20bc8bf/torchvision/datasets/mnist.py#L291
```suggestion
```

Reply:
@akihironitta You are correct that torchvision does run a check. It came to my mind while writing this method, but I kept it there as an additional fail-safe. See, the dataset is not called before the `prepare()` and then the `setup()` methods. These methods are also not called at the time of instantiation of the datamodule class. So, technically, one could check `num_classes` property even before calling (fetching) the dataset. The `_emnist_split_exists()` ensures that even in such cases, the code does not do something odd.

Reply:
:bulb: Given that the dependency of `num_classes` has been lifted from EMNIST_METADATA, I think this method `_emnist_split_exists()` could be removed as well.

Reply:
@sugatoray To me, it' looks just redundant since it's something that's been done already in torchvision. It's up to you then, but other maintainers might suggest removing it later.

Reply:
@akihironitta Thank you. I am keeping this as a non-resolved conversation to get back to this later on. Thank you, for all your reviews. Let me attend to them first, and then we can decide what to do about this later.